github/github.go

```diff
@@ -346,6 +347,19 @@ func NewClient(httpClient *http.Client) *Client {
 	return c
 }
 
+// NewClient returns a new GitHub API client authenticated with the provided token.
+func NewTokenClient(ctx context.Context, token string) *Client {
+	return NewClient(
+		oauth2.NewClient(
+			ctx, oauth2.StaticTokenSource(
+				&oauth2.Token{
+					AccessToken: token,
+				},
+			),
+		),
```

Comment:
How about collapsing these 7 lines?

```suggestion
		oauth2.NewClient(ctx, oauth2.StaticTokenSource(	&oauth2.Token{AccessToken: token})),
```